__TESTS__/unit/analytics/analytics.node.test.ts

```diff
@@ -169,4 +169,40 @@ describe('Add analytics to a regular URL', () => {
       }
     })).toBe("https://res.cloudinary.com/demo/image/upload/sample%3Fparam%3Dtest");
   });
+
+  it('Should not include analytics when authToken is provided', () => {
+    const cldImage = createNewImageWithAnalytics('sample');
+    cldImage.sign();
```

Comment:
Why is the sign() function needed?

Reply:
It's a way for the user to explicitly state he wants to sign the URL, instead of us implicitly assuming he wants to because he provided a signature or a token.